packages/workflow/src/records/fhir.ts

```diff
@@ -103,21 +113,221 @@ export function createCorrectionProofOfLegalCorrectionDocument(
+    return [
+      {
+        fullUrl: `urn:uuid:${temporaryRelatedPersonId}`,
+        resource: {
+          resourceType: 'RelatedPerson',
+          relationship: {
+            coding: [
+              {
+                system:
+                  'http://hl7.org/fhir/ValueSet/relatedperson-relationshiptype',
+                code: collectorDetails.relationship
+              }
+            ],
+            text: collectorDetails.otherRelationship
+          },
+          ...(collectorDetails.affidavit?.[0] && {
+            extension: [
+              {
+                url: `http://opencrvs.org/specs/extension/relatedperson-affidavittype`,
+                valueAttachment: {
+                  ...collectorDetails.affidavit[0]
+                }
+              }
+            ]
+          }),
+          patient: {
+            reference: `urn:uuid:${temporaryPatientId}`
+          }
+        }
+      },
+      {
+        fullUrl: `urn:uuid:${temporaryPatientId}`,
+        resource: {
+          resourceType: 'Patient',
+          name: collectorDetails.name.map(
+            ({ use, firstNames, familyName }) => ({
+              use,
+              given: firstNames.split(' '),
+              family: [familyName]
+            })
+          ),
+          identifier: collectorDetails.identifier.map(({ id, type }) => ({
+            id,
+            type: {
+              coding: [
+                {
+                  system: 'http://opencrvs.org/specs/identifier-type',
+                  code: type
+                }
+              ]
+            }
+          }))
+        }
+      }
+    ]
+  }
+
+  const parseResult = knownRelationships.safeParse(
+    collectorDetails.relationship
+  )
+
+  if (parseResult.success) {
+    const knownRelationship = parseResult.data
+    const section = findCompositionSection(
+      relationshipToSectionCode(knownRelationship),
+      getComposition(record)
+    )
+
+    if (!section) {
+      throw badRequest(`Patient resource for ${knownRelationship} not found`)
+    }
+
+    return [
+      {
+        fullUrl: `urn:uuid:${temporaryRelatedPersonId}`,
+        resource: {
+          resourceType: 'RelatedPerson',
+          relationship: {
+            coding: [
+              {
+                system:
+                  'http://hl7.org/fhir/ValueSet/relatedperson-relationshiptype',
+                code: collectorDetails.relationship
+              }
+            ]
+          },
+          patient: {
+            reference: section.entry[0].reference
+          }
+        }
+      }
+    ]
+  }
+
+  return [
+    {
+      fullUrl: `urn:uuid:${temporaryRelatedPersonId}`,
+      resource: {
+        resourceType: 'RelatedPerson',
+        relationship: {
+          coding: [
+            {
+              system:
+                'http://hl7.org/fhir/ValueSet/relatedperson-relationshiptype',
+              code: collectorDetails.relationship
+            }
+          ]
+        }
+      }
+    }
+  ]
+}
```

Comment:
Not sure when we need no patient reference in the RelatedPerson entry. Since you're already handling the case of `otherRelationship`. Could be also reasonable to leave it as is for corner cases maybe?

Reply:
As far as I remember, that case is for collectors about whom we don't have any details in our form e.g. "Legal Guardian". I'll add a comment there